app/aircopy.c

```diff
@@ -60,170 +70,172 @@ void AIRCOPY_SendMessage(void)
 
 	gAircopySendCountdown = 30;
+
+	return 0;
 }
 
 void AIRCOPY_StorePacket(void)
 {
-	uint16_t Status;
-
-	if (gFSKWriteIndex < 36)
+	if (gFSKWriteIndex < 36) {
 		return;
+	}
 
 	gFSKWriteIndex = 0;
 	gUpdateDisplay = true;
-	Status         = BK4819_ReadRegister(BK4819_REG_0B);
+	uint16_t Status = BK4819_ReadRegister(BK4819_REG_0B);
 	BK4819_PrepareFSKReceive();
 
 	// Doc says bit 4 should be 1 = CRC OK, 0 = CRC FAIL, but original firmware checks for FAIL.
 
-	if ((Status & 0x0010U) == 0 && g_FSK_Buffer[0] == 0xABCD && g_FSK_Buffer[35] == 0xDCBA)
-	{
-		uint16_t     CRC;
-		unsigned int i;
-
-		for (i = 0; i < 34; i++)
-			g_FSK_Buffer[i + 1] ^= Obfuscation[i % 8];
-
-		CRC = CRC_Calculate(&g_FSK_Buffer[1], 2 + 64);
-		if (g_FSK_Buffer[34] == CRC)
-		{
-			const uint16_t *pData;
-			uint16_t        Offset;
-
-			Offset = g_FSK_Buffer[1];
-			if (Offset < 0x1E00)
-			{
-				pData = &g_FSK_Buffer[2];
-				for (i = 0; i < 8; i++)
-				{
-					EEPROM_WriteBuffer(Offset, pData);
-					pData  += 4;
-					Offset += 8;
-				}
-				
-				if (Offset == 0x1E00)
-					gAircopyState = AIRCOPY_COMPLETE;
-
-				gAirCopyBlockNumber++;
-
-				return;
-			}
-		}
+	if ((Status & 0x0010U) != 0 || g_FSK_Buffer[0] != 0xABCD || g_FSK_Buffer[35] != 0xDCBA) {
+		gErrorsDuringAirCopy++;
+		return;
+	}
+
+	for (unsigned int i = 0; i < 34; i++) {
+		g_FSK_Buffer[i + 1] ^= Obfuscation[i % 8];
 	}
-	gErrorsDuringAirCopy++;
+
+	uint16_t CRC = CRC_Calculate(&g_FSK_Buffer[1], 2 + 64);
+	if (g_FSK_Buffer[34] != CRC) {
+		gErrorsDuringAirCopy++;
+		return;
+	}
+
+	uint16_t Offset = g_FSK_Buffer[1];
+
+	if (Offset >= 0x1E00) {
+		gErrorsDuringAirCopy++;
+		return;
+	}
+
+	const uint16_t *pData = &g_FSK_Buffer[2];
+	for (unsigned int i = 0; i < 8; i++) {
+		EEPROM_WriteBuffer(Offset, pData);
+		pData += 4;
+		Offset += 8;
+	}
+
+	if (Offset == 0x1E00) {
+		gAircopyState = AIRCOPY_COMPLETE;
+	}
+
+	gAirCopyBlockNumber++;
 }
 
 static void AIRCOPY_Key_DIGITS(KEY_Code_t Key, bool bKeyPressed, bool bKeyHeld)
 {
-	if (!bKeyHeld && bKeyPressed)
-	{
-		uint32_t      Frequency;
-		unsigned int  i;
-
-		INPUTBOX_Append(Key);
-		gRequestDisplayScreen = DISPLAY_AIRCOPY;
-		if (gInputBoxIndex < 6)
-		{
-			#ifdef ENABLE_VOICE
-				gAnotherVoiceID = (VOICE_ID_t)Key;
-			#endif
-			return;
+	if (bKeyHeld || !bKeyPressed) {
+		return;
+	}
+
+	INPUTBOX_Append(Key);
+
+	gRequestDisplayScreen = DISPLAY_AIRCOPY;
+
+	if (gInputBoxIndex < 6) {
+#ifdef ENABLE_VOICE
+		gAnotherVoiceID = (VOICE_ID_t)Key;
+#endif
+		return;
+	}
+
+	gInputBoxIndex = 0;
+	uint32_t Frequency = StrToUL(INPUTBOX_GetAscii()) * 100;
+
+	for (unsigned int i = 0; i < BAND_N_ELEM; i++) {
+		if (Frequency < frequencyBandTable[i].lower || Frequency >= frequencyBandTable[i].upper) {
+			continue;
 		}
 
-		gInputBoxIndex = 0;
-		Frequency = StrToUL(INPUTBOX_GetAscii()) * 100;	
-
-		for (i = 0; i < BAND_N_ELEM; i++)
-		{
-			if (Frequency >= frequencyBandTable[i].lower && Frequency < frequencyBandTable[i].upper)
-			{
```

Comment:
There was not TX restriction check, just if the frequency belonged to some band.